src/shared/user-record.c

```diff
@@ -1871,6 +1874,15 @@ uint64_t user_record_luks_pbkdf_parallel_threads(UserRecord *h) {
         return MIN(h->luks_pbkdf_parallel_threads, UINT32_MAX);
 }
 
+uint64_t user_record_luks_sector_size(UserRecord *h) {
+        assert(h);
+
+        if (h->luks_sector_size == UINT64_MAX)
+                return 512;
+
+        return MIN(h->luks_sector_size, UINT32_MAX);
```

Comment:
i'd probably mangle things further here, i.e. round down to closest power of two, round up to minimum of 512. 

i.e. i think user facing it's good to generate proper error messages for obviously invalid parameters. but on the lower-level, non-user-facing codepaths (and the json stuff qualifies) I'd mangle things so that they roughly make sense.


```c
if (h->luks_sector_size <= 512 || h->luks_sector_size == UINT64_MAX)
        return 512;

return CLAMP(UINT64_C(1) << (63 - __builtin_clzl(h->luks_sector_size)), 512, UINT32_C(0x80000000));
```